### DISCUSSION TITLE
How is i_CTRL-O changed in Lazyvim?

### DISCUSSION THREAD

**Discussion by shenyfg** [LazyVim/LazyVim#4368] [Q&A] [answered]
By default, i_CTRL-O is used to temparily switch to normal mode when inserting, while in lazyvim, it just simply **return to the normal mode**. And I can' figure out how it was changed, as well as how to restore it to default.

I've tried this in the lazyvim's docker, just same as my own configuration. Any help?

_ps: the description in vim's doc/insert.txt_
`CTRL-O		execute one command, return to Insert mode   *i_CTRL-O*`

**Comment by dpetka2001** [comment]
LazyVim doesn't change anything about `i_Ctrl-O`. It works correctly with a default LazyVim installation. Must be some kind of configuration on your end or maybe an issue with how your terminal interprets the key sequence.

**Reply by shenyfg** [reply]
I do use alacritty as my terminal, and tmux to manage my sessions, while I'm pretty sure there are not any configs for CTRL-O. 

On the other hand, **`nvim -u NONE` works correctly for me**, as well as `vim -u NONE`. However, in lazyvim, both my configuration and the default lazyvim docker image would turn `i_CTRL_O` to `<ESC>`.

Does your machine work correctly?

**Reply by dpetka2001** [reply, marked as answer]
Here is a screencast of installing default LazyVim from scratch in kitty terminal. As you can see it works correctly

[Screencast 2024-08-29 16:32:11.webm](https://github.com/user-attachments/assets/48066872-c3d2-4fba-a920-b807d5fda035)

Also what does `:verbose imap <C-o>` show?

**Reply by shenyfg** [reply]
Oh, I figured out the problem. It is not a issue actually, just different ways of showing **VIM MODE** between plain nvim and the plugin of **lualine**.

In lualine, when I type CTRL-O in insert mode, the mode just temporily turns to `NORMAL` mode. While in plain nvim, it switches from `INSERT` to `(insert)`(see the pics I show below, a rare mode I haven't come across before), which has puzzled me. In fact, it turns out they work the same.

Huge thanks for your patient replies, I'll mark as answer.

<img width="421" alt="image" src="https://github.com/user-attachments/assets/8f0cc112-a719-4e17-bdd6-eeaeac0cefef">
<img width="455" alt="image" src="https://github.com/user-attachments/assets/04f6d17e-32cf-4876-87f4-9ab48a25c081">
